src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -754,14 +760,48 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
         if (ptr.PointerDeviceType() == Windows::Devices::Input::PointerDeviceType::Mouse)
         {
-            if (point.Properties().IsLeftButtonPressed())
+            if (_terminal->IsSelectionActive() && point.Properties().IsLeftButtonPressed())
             {
                 const auto cursorPosition = point.Position();
-                const auto terminalPosition = _GetTerminalPosition(cursorPosition);
+                _SetEndSelectionPointAtCurstor(cursorPosition);
 
-                // save location (for rendering) + render
-                _terminal->SetEndSelectionPosition(terminalPosition);
-                _renderer->TriggerSelection();
+                const double cursorBelowBottomDist = cursorPosition.Y - _root.Padding().Top - _swapChainPanel.ActualHeight();
+                const double cursorAboveTopDist = -cursorPosition.Y + _root.Padding().Top;
+                constexpr double distThreashold = 2.0;
+
+                if (cursorBelowBottomDist > distThreashold)
+                {
+                    _autoScrollVelocity = _GetAutoScrollSpeed(cursorBelowBottomDist);
+                }
+                else if (cursorAboveTopDist > distThreashold)
+                {
+                    _autoScrollVelocity = -_GetAutoScrollSpeed(cursorAboveTopDist);

```

Comment:
Oho! There's that sneaky `-` again. Mind making it less sneaky and writing something like `(-1.0)*` instead.